### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by choldgraf**
This just adds a few pre-commit hooks and runs them. The biggest change is probably `mdformat` - seems reasonable just to standardize our markdown structure, but I don't have a strong opinion if others don't want to use that.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/team-meeting.md [hunk 1]
```diff
@@ -16,11 +12,11 @@ If you'd like to discuss something at the meeting, please add an item to the age
 
 ### Before the meeting
 
-- [ ] Update dates and make sure HackMD information is correct.
-- [ ] Team members add agenda items.
-  
+- \[ \] Update dates and make sure HackMD information is correct.
```
Comment (chrisjsewell) [inline .github/ISSUE_TEMPLATE/team-meeting.md:15]: broken the todo lists

File: docs/feature-vote.md [hunk 2]
```diff
@@ -1,12 +1,13 @@
 (feature-note)=
+
 # Feature voting board
 
 You can help us to prioritise development of new features by leaving a [👍 reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) on the first comment of any `enhancement` issue.
 
-Below is a list of all current enhancement issues from our core repositories,[^a] ordered by 👍
+Below is a list of all current enhancement issues from our core repositories,\[^a\] ordered by 👍
```
Comment (chrisjsewell) [inline docs/feature-vote.md:7]: broken the footnotes

File: .github/ISSUE_TEMPLATE/team-meeting.md [hunk 3]
```diff
@@ -1,10 +1,6 @@
----
```
Comment (chrisjsewell) [inline .github/ISSUE_TEMPLATE/team-meeting.md:1]: broken the front matter

File: docs/feature-vote.md (hunk 2 above)
Comment (choldgraf) [reply docs/feature-vote.md:7]: argh - hmm what should we do about these? add exceptions in the mdformat config or something?

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  - repo: https://github.com/psf/black
+    rev: 22.1.0
+    hooks:
+    - id: black
+
+  - repo: https://github.com/executablebooks/mdformat
+    rev: 0.7.13
+    hooks:
+      - id: mdformat
```
Comment (chrisjsewell) [inline .pre-commit-config.yaml:27]: yeh so you need to use this with plugins 😉 e.g.

- https://github.com/executablebooks/mdformat-myst
- https://github.com/hukkin/mdformat-gfm (for tasklists)

File: docs/feature-vote.md (hunk 2 above)
Comment (chrisjsewell) [reply docs/feature-vote.md:7]: plugins 👇

File: .pre-commit-config.yaml (hunk 4 above)
Comment (choldgraf) [reply .pre-commit-config.yaml:27]: ahh ok, I will try to give that a shot

Comment (choldgraf) [reply .pre-commit-config.yaml:27]: ahhhh, ok I will take a look and figure it out

Comment (chrisjsewell) [reply .pre-commit-config.yaml:27]: Meaning to circle back to mdformat-myst at some point and do some more testing/improvements.
Would be nice to advertise it on jupyter-book (but only when I'm certain it is properly working)

File: docs/index.md [hunk 5]
```diff
@@ -7,54 +7,54 @@ computational narratives using the Jupyter ecosystem.
 
 We are the community behind [Jupyter Book](https://jupyterbook.org)--an open source project for building beautiful, publication-quality books and documents from computational material--as well as [MyST Markdown](https://myst-parser.readthedocs.io/en/latest/)--a rich and extensible flavor of Markdown meant for technical documentation and publishing.
 
-::::{grid} 1 2 2 2
+::::\{grid} 1 2 2 2
```
Comment (choldgraf) [inline docs/index.md:10]: I updated the PR to add those plugins to our hook.

Hmm - looks like it is not picking up the colon fence syntax, is there a way to tell the pre-commit hook to re-use the same plugins that are enabled in `conf.py`?

Comment (chrisjsewell) [reply docs/index.md:10]: https://github.com/executablebooks/mdformat-myst/issues/13 and https://github.com/executablebooks/mdformat-myst/pull/9

But no, not presently 😢

Comment (choldgraf) [reply docs/index.md:10]: ah bummer - ok I'll just comment out the lines about mdformat and we can re-enable if/when there is support for extra myst extensions.

**Comment by choldgraf** [comment]
OK latest commit comments out the mdformat hook, so we're just doing some basic code and file standardization stuff here

**Comment by choldgraf** [comment]
closing this one as there are some conflicts and I don't have time to get it over the finish line

**Pull request closed without merging**